### PR TITLE
Daos drain master change

### PR DESCRIPTION
**Pull request by wangdi1**
DAOS-10525 pool: update dom status during reintegration.
Only update domain status if the original status is DOWN
or DOWNOUT, i.e. partial targets reintegration may not
need change domain status.

Signed-off-by: Di Wang <di.wang@intel.com>

### CODE REVIEW

File: src/rebuild/srv.c [hunk 1]
```diff
@@ -1224,7 +1299,7 @@ rebuild_task_ult(void *arg)
 	}
 
 	/* Wait until rebuild finished */
-	rebuild_leader_status_check(pool, task->dst_map_ver,
+	rebuild_leader_status_check(pool, task->dst_map_ver, task->dst_map_ver,
```
Comment (liuxuezhao) [inline src/rebuild/srv.c:1302]: 3rd parameter is a typo of "task->dst_max_ver"?

File: src/rebuild/srv.c [hunk 2]
```diff
@@ -595,44 +601,42 @@ rebuild_leader_status_check(struct ds_pool *pool, uint32_t map_ver, uint32_t op,
...
-				" %u->%u.\n", DP_UUID(pool->sp_uuid), myrank,
-				pool->sp_iv_ns->iv_master_rank);
+			dom = pool_map_find_node_by_rank(pool->sp_map, excluded.rl_ranks[i]);
+			D_ASSERT(dom != NULL);
+
+			/* If the rank is being scheduled after reintegration
+			 * start, let's skip the rank, i.e. set it DONE.
+			 */
+			if (dom->do_comp.co_status == PO_COMP_ST_UP &&
+			    dom->do_comp.co_in_ver < max_ver)
```
Comment (liuxuezhao) [inline src/rebuild/srv.c:630]: should above line be "<="?

File: src/rebuild/srv.c [hunk 3]
```diff
@@ -790,16 +795,12 @@ rebuild_prepare(struct ds_pool *pool, uint32_t rebuild_ver,
...
 	pool_comp_state_t	match_status;
-	unsigned int		master_rank;
 	int			rc;
 
 	D_DEBUG(DB_REBUILD, "pool "DF_UUID" create rebuild iv, op=%s\n",
 		DP_UUID(pool->sp_uuid), RB_OP_STR(rebuild_op));
 
 	/* Update pool iv ns for the pool */
-	crt_group_rank(pool->sp_group, &master_rank);
-	ds_pool_iv_ns_update(pool, master_rank);
```
Comment (liuxuezhao) [inline src/rebuild/srv.c:801]: sorry, why drop the code? need not update the master_rank?

File: src/rebuild/rebuild_internal.h [hunk 4]
```diff
@@ -203,6 +206,7 @@ struct rebuild_task {
 	uint64_t			dst_schedule_time;
 	uint32_t			dst_map_ver;
 	uint32_t			dst_rebuild_gen;
+	uint32_t			dst_max_ver;
```
[the remark this answers is not in the document]
Comment (wangdi1) [reply src/rebuild/rebuild_internal.h:209]: sure

File: src/rebuild/srv.c (hunk 1 above)
Comment (wangdi1) [reply src/rebuild/srv.c:1302]: ah, right.

File: src/rebuild/srv.c (hunk 2 above)
Comment (wangdi1) [reply src/rebuild/srv.c:630]: ah, right.

File: src/rebuild/srv.c (hunk 3 above)
Comment (wangdi1) [reply src/rebuild/srv.c:801]: oh, iv ns update should be done through iv pool map update, instead of by rebuild iv, just cleanup a bit.

**Comment by wangdi1** [comment]
the patch has been landed by another PR-9744

**Pull request closed without merging**